radixdlt/babylon-wallet-ios
# ABW-1752 Implement ROLA prefix

Jira ticket: https://radixdlt.atlassian.net/browse/ABW-1752

Complements / depends on / relates to #link to other PR

## Description
It adds the "R" (`0x52`) prefix to the ROLA payload. More details in the description and linked discussion in [ABW-1751](https://radixdlt.atlassian.net/browse/ABW-1751). It was necessary to recreate test vectors. [They should be in line with what we have on the Android side](https://rdxworks.slack.com/archives/C031A0V1A1W/p1688988182684739). Here's the [related PR on Android side](https://github.com/radixdlt/babylon-wallet-android/pull/325).

## How to test

Use [v2-compatible Sandbox](https://radix-dapp-toolkit-pr-51.rdx-works-main.extratools.works/), for example https://radix-dapp-toolkit-pr-51.rdx-works-main.extratools.works/
1. Link wallet and connect to dApp
2. In ROLA section verify both "Send login request" and "Verify accounts" paths - in both cases you should see the `"verified": true` in logs.

## PR submission checklist


[ABW-1751]: https://radixdlt.atlassian.net/browse/ABW-1751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
